src/shogun/mathematics/linalg/internal/implementation/Scale.h

```diff
@@ -66,6 +66,41 @@ struct scale
 	static void compute(Matrix A, Matrix B, Matrix C, T alpha, T beta);
 };
 
+/** Specialization of scale for the Native backend */
+template<class Matrix>
+struct scale<Backend::NATIVE, Matrix>
+{
+	typedef typename Matrix::Scalar T;
+	
+	/** Performs the operation B = alpha*A */
+	static void compute(SGMatrix<T> A, SGMatrix<T> B, T alpha)
+	{
+		REQUIRE((A.num_rows==B.num_rows)&&(A.num_cols==B.num_cols),
+			"Dimensions of A(%dx%d) and B(%dx%d) don't match.\n",
```

Comment:
@iglesias I personally like the `\n` in the end - as per our wiki on [Assertions] -(https://github.com/shogun-toolbox/shogun/wiki/Assertions).

- Use **proper** English (Start with capital letters, write sentences)
- Don't forget `\n` at the end.
- Make sure that you **don't cause segfaults** in your assertions, check bounds, pointer valid, etc

Although it is just a matter of choice.